app/lib/tokumei/close_connection.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Tokumei.CloseConnection do
+  @moduledoc """
+  If we respond to a request with Connection: close, we should send a response
+  back with Connection: close and then close it ourselves.
+
+  This works around buggy clients that send Connection: close and never close the
+  connection themselves.
+  """
+
+  defmacro __using__(_opts) do
+    quote do
+      @before_compile unquote(__MODULE__)
+    end
+  end
+  defmacro __before_compile__(_env) do
+    quote location: :keep do
+      defoverridable [handle_request: 2]
+
+      def handle_request(request, env) do
+        response = super(request, env)
+
+        response_headers = case request.headers do
+          %{"connection" => "close"} -> [{"connection", "close"} | response.headers]
```

Comment:
Ok, but in this case you are matching on a map of headers and adding to a tuple list of response headers so one is incorrect. Also a little embarassing but I think my docs are lying to you. we chatted about this a lot on an issue previously https://github.com/CrowdHailer/raxx/issues/13. I would just put some inspects in somewhere and see what it really is